### PR TITLE
fix #10 undo the genral py.test hack

### LINKED ISSUE #10: the `py.test workaround hides legitimate mistakes
c9b997713cb77d2c1334acb7847ee6b24e3261b2 introduced returning None on missing aliasmodules unconditionally
this does hide actual errors

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
add a opt-in
add a auto-opt-in for the py.test module

fixes #10

### CODE REVIEW

**Review by RonnyPfannschmidt — COMMENTED** [review]
Im going to make this one not exposed after all

Just going to add a hack only to pylib

File: src/apipkg/__init__.py [hunk 1]
```diff
@@ -180,6 +180,24 @@ def __dict__(self):
         return dict
 
 
+def _py_test_hack(modpath, attrname, importerror_alternative):
```
Comment (nicoddemus) [inline src/apipkg/__init__.py:183]: Probably good to leave a docstring here explaining what is the hack and why it is necessary.

File: src/apipkg/__init__.py [hunk 2]
```diff
@@ -191,18 +209,21 @@ def getmod():
             mod.append(x)
         return mod[0]
 
+    repr_result = _alias_mod_repr(modname, modpath, attrname)
```
Comment (nicoddemus) [inline src/apipkg/__init__.py:212]: Probably inlining this will make things simpler.

File: src/apipkg/__init__.py (hunk 1 above)
Comment (nicoddemus) [reply src/apipkg/__init__.py:183]: Actually I noticed this is not being used anywhere... is that intentional?

File: test_apipkg.py [hunk 3]
```diff
@@ -551,11 +551,25 @@ def test_aliasmodule_aliases_an_attribute():
     assert not hasattr(am, "lqkje")
 
 
-def test_aliasmodule_aliases_unimportable():
+def test_aliasmodule_aliases_unimportable_fails():
     am = apipkg.AliasModule("mymod", "qlwkejqlwe", "main")
     r = repr(am)
     assert "<AliasModule 'mymod' for 'qlwkejqlwe.main'>" == r
-    assert am.qwe is None
+    # this would pass starting with apipkg 1.3 to work around a pytest bug
```
Comment (nicoddemus) [inline test_apipkg.py:558]: Which bug? Do we have an issue to link? (the answers could be in a comment in the code)

**Review by nicoddemus — COMMENTED** [review]
I'm not sure what the change is about, TBH, but I left a few comments. 👍

File: src/apipkg/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/apipkg/__init__.py:183]: missed code drop from the simplification

File: src/apipkg/__init__.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply src/apipkg/__init__.py]: don

File: test_apipkg.py (hunk 3 above)
Comment (RonnyPfannschmidt) [reply test_apipkg.py:558]: #10

File: src/apipkg/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/apipkg/__init__.py]: removed

**Pull request merged**